hyperdash/hyper_dash.py

```diff
@@ -296,36 +296,36 @@ def network_loop():
                     self.server_manager.tick(self.current_sdk_run_uuid)
                     time.sleep(1)
 
-        code_thread = Thread(target=self.code_runner.run)
         network_thread = Thread(target=network_loop)
-
-        # Daemonize them so they don't impede shutdown if the user
+        # Daemonize so they don't impede shutdown if the user
         # keyboard interrupts
-        code_thread.daemon = True
         network_thread.daemon = True
-
         network_thread.start()
-        code_thread.start()
 
+        #Create thread for running code if using CLI or decorator
+        if type(self.runner) is CodeRunner:
```

Comment:
Ah....I really don't like that you're checking the type here....Can you add a method to the CodeRunners thats like "should_run_as_thread" or something so that the injected object can just tell it if it needs to be run as a thread or not

Reply:
Good call.